contents/blog/improving-posthog-deployments.md

```diff
@@ -0,0 +1,99 @@
+
+The more simple the setup and maintenance, the lower the barrier of entry, and the more feedback we received on where to take the product.
+
+TODO: <insert diagram of event data flow from [here](https://www.google.com/url?q=https://posthog.com/handbook/engineering/databases/event-ingestion&sa=D&source=docs&ust=1647446368828545&usg=AOvVaw31KohI1YJiLlpt2SX8wP8P)>
+
+As time went by, the number of customers and the volume of data to ingest and analyze increased, and we realized that this simple architecture was not scalable. Our event queries using PostgreSQL were slow, and our ingestion couldn’t keep up with the volumes of events we were receiving. These limitations led us to revisit our stack:
+
+TODO: <insert slowly moving hedgehog image>
+
+Introducing Kafka helped us to decouple our data streams and our processing pipeline, while ClickHouse helped us to deliver very fast queries on a large volume of data and more horizontal scaling capabilities (for more detailed information see the [ClickHouse announcement](https://posthog.com/blog/clickhouse-announcement)). While these additions helped us scale, they come at a cost:
+
+* We now have additional distributed services like Kafka, ClickHouse and Zookeeper to deploy and manage.
+
+* Each additional dependency requires considerations in terms of monitoring, migrations, upgrades, durability among others.
+
+* ClickHouse is powerful but relatively new compared to PostgreSQL, very few people know or have experience with managing a ClickHouse cluster.
+
+* ClickHouse is a relatively new product. Whereas tooling around PostgreSQL for example is very mature, we need to do some of the heavy lifting ourselves with ClickHouse.
+
+PostHog is built on the idea that anyone should be able to deploy, maintain and use it. We may be experts, but our users should not need to be.
+
+Adding distributed systems to our stack also increased its complexity (we also had to bring 3rd party dependencies like Zookeeper) and what was initially an architecture easy to understand, deploy and maintain suddenly became more complex. Simple operations such as application deployments or database upgrades have now become complex automations handling distributed systems. Supporting self-hosted and non-self-hosted installations without increasing complexity has thus become a major challenge.
+
+To self-host PostHog, it is still possible to use docker-compose for the evaluation stage, but what about the move into production? We currently offer an abstraction system built on top of the [Kubernetes](https://kubernetes.io/) platform via [Helm](https://helm.sh/). We’ve built on top of Kubernetes because although for many scenarios it is overkill, it allows us to focus on one abstraction but target any cloud provider that has a Kubernetes offerings. Currently we support AWS, Azure (alpha), Digital Ocean and Google Cloud Platform.
```

Comment:
This section kind of breaks the flow a bit at the moment, but adds useful context for techincal users. Maybe we could put this in a boxout as an aside, or under its own heading?

Reply:
Done!